triton-docker

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+set -e
+
+# Check if triton is installed and executable
+function checkTritonInstalled () {
+    command -v triton >/dev/null 2>&1 || { echo >&2 "Triton CLI tools do not appear to be installed. Please install before continuing."; exit 1; }
+}
+
+# Check if curl is installed and executable
+function checkCurlInstalled () {
+    command -v curl >/dev/null 2>&1 || { echo >&2 "curl is required to complete this installation. Please install before continuing."; exit 1; }
+}
+
+# Emit the current Triton profile and time
+function currentProfile () {
+    if [ -n "$TRITON_PROFILE" ]
+    then
+        local profile_source='via env var; use `eval "$(triton env <profile name>)"` to change'
+    else
+        local profile_source='default; use `triton profile set <profile name>` to change'
+    fi
+
+    echo "Executing in '$(triton profile get | awk 'NR == 1 {print $2}')' (${profile_source}) at $(date +'%r')"
+}
+
+# Check if the Docker binary (named triton-docker-helper) is installed and executable
+# ...install it if not
+function checkDockerInstalled () {
+    if [ ! -f /usr/local/bin/triton-docker-helper ] || [ ! -x /usr/local/bin/triton-docker-helper ]
+    then
+        echo "Triton-docker needs to install an application to continue."
+        echo "This installation will only happen once."
+
+        # Check if triton is installed and executable
+        command -v curl >/dev/null 2>&1 || { echo >&2 "curl is required to complete this installation. Please install before continuing."; exit 1; }
+
+        rm -Rf /tmp/triton-docker /usr/local/bin/triton-docker-helper
+        mkdir -p /tmp/triton-docker
+        curl https://get.docker.com/builds/$(uname -a | awk '{ print $1 }')/x86_64/docker-1.12.6.tgz | tar zxvf - -C /tmp/triton-docker
+        mv /tmp/triton-docker/docker/docker /usr/local/bin/triton-docker-helper
+        rm -Rf /tmp/triton-docker
+        chmod +x /usr/local/bin/triton-docker-helper
+
+        echo "The triton-docker-helper is now installed."
+        echo
+    fi
+}
+
+# Check if Docker Compose (named triton-compose-helper) is installed and executable
+# ...install it if not
+function checkComposeInstalled () {
+    if [ ! -f /usr/local/bin/triton-compose-helper ] || [ ! -x /usr/local/bin/triton-compose-helper ]
+    then
+        echo "Triton-compose needs to install an application to continue."
```

Comment:
Is there any chance this going to require sudo? I feel like it might be better to fail and then ask the user to run an install rather than going ahead, particularly if we're not telling them just what we're installing.